File: app/src/main/java/com/example/ui/components/OrderItemNotesDialog.kt
package com.example.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FilterChip
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.CartItem
import com.example.ui.theme.AmberPrimary

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun OrderItemNotesDialog(
    item: CartItem,
    onSaveNotes: (String) -> Unit,
    onDismiss: () -> Unit
) {
    var noteText by remember { mutableStateOf(item.notes) }

    val presetNotes = listOf(
        "Pedas Sedang",
        "Extra Pedas 🔥",
        "Tidak Pedas",
        "Tanpa Bawang",
        "Sedikit Minyak",
        "Es Dipisah",
        "Less Sugar (50%)",
        "Bungkus Terpisah"
    )

    AlertDialog(
        onDismissRequest = onDismiss,
        title = {
            Column {
                Text(
                    text = "Catatan Pesanan",
                    fontWeight = FontWeight.Bold,
                    fontSize = 18.sp
                )
                Text(
                    text = item.menu.name,
                    fontSize = 13.sp,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        },
        text = {
            Column(modifier = Modifier.fillMaxWidth()) {
                OutlinedTextField(
                    value = noteText,
                    onValueChange = { noteText = it },
                    placeholder = { Text("Contoh: Pedas sedang, jangan pakai kecap...") },
                    modifier = Modifier
                        .fillMaxWidth()
                        .testTag("item_notes_input"),
                    shape = RoundedCornerShape(12.dp),
                    minLines = 2,
                    maxLines = 3
                )

                Spacer(modifier = Modifier.height(12.dp))

                Text(
                    text = "Catatan Cepat:",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )

                Spacer(modifier = Modifier.height(6.dp))

                Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    val chunked = presetNotes.chunked(2)
                    chunked.forEach { rowPresets ->
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.spacedBy(6.dp)
                        ) {
                            rowPresets.forEach { preset ->
                                FilterChip(
                                    selected = noteText.contains(preset),
                                    onClick = {
                                        noteText = if (noteText.isBlank()) preset else "$noteText, $preset"
                                    },
                                    label = { Text(preset, fontSize = 11.sp) },
                                    modifier = Modifier.weight(1f)
                                )
                            }
                        }
                    }
                }
            }
        },
        confirmButton = {
            Button(
                onClick = { onSaveNotes(noteText.trim()) },
                colors = ButtonDefaults.buttonColors(containerColor = AmberPrimary),
                shape = RoundedCornerShape(10.dp),
                modifier = Modifier.testTag("save_notes_btn")
            ) {
                Text("Simpan Catatan")
            }
        },
        dismissButton = {
            OutlinedButton(
                onClick = onDismiss,
                shape = RoundedCornerShape(10.dp)
            ) {
                Text("Batal")
            }
        },
        shape = RoundedCornerShape(20.dp)
    )
}

File: lib/widgets/order_item_notes_dialog.dart
import 'package:flutter/material.dart';
import '../models/cart_item_model.dart';
import '../theme/app_colors.dart';

class OrderItemNotesDialog extends StatefulWidget {
  final CartItemModel item;
  final ValueChanged<String> onSaveNotes;
  final VoidCallback onDismiss;

  const OrderItemNotesDialog({
    super.key,
    required this.item,
    required this.onSaveNotes,
    required this.onDismiss,
  });

  @override
  State<OrderItemNotesDialog> createState() => _OrderItemNotesDialogState();
}

class _OrderItemNotesDialogState extends State<OrderItemNotesDialog> {
  late TextEditingController _noteController;

  final List<String> _presetNotes = const [
    'Pedas Sedang',
    'Extra Pedas 🔥',
    'Tidak Pedas',
    'Tanpa Bawang',
    'Sedikit Minyak',
    'Es Dipisah',
    'Less Sugar (50%)',
    'Bungkus Terpisah',
  ];

  @override
  void initState() {
    super.initState();
    _noteController = TextEditingController(text: widget.item.notes);
  }

  @override
  void dispose() {
    _noteController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Dialog(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
      backgroundColor: Colors.white,
      child: Padding(
        padding: const EdgeInsets.all(18),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              'Catatan Pesanan',
              style: const TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
            ),
            Text(
              widget.item.menu.name,
              style: const TextStyle(fontSize: 12, color: AppColors.textSecondaryLight),
            ),
            const SizedBox(height: 12),

            TextField(
              controller: _noteController,
              maxLines: 2,
              decoration: InputDecoration(
                hintText: 'Contoh: Pedas sedang, jangan pakai kecap...',
                hintStyle: const TextStyle(fontSize: 12, color: AppColors.textSecondaryLight),
                border: OutlineInputBorder(borderRadius: BorderRadius.circular(12)),
                contentPadding: const EdgeInsets.all(12),
              ),
            ),

            const SizedBox(height: 12),

            const Text(
              'Catatan Cepat:',
              style: TextStyle(fontSize: 11, fontWeight: FontWeight.bold, color: AppColors.textSecondaryLight),
            ),
            const SizedBox(height: 6),

            Wrap(
              spacing: 6,
              runSpacing: 4,
              children: _presetNotes.map((preset) {
                final isSelected = _noteController.text.contains(preset);
                return FilterChip(
                  selected: isSelected,
                  onSelected: (_) {
                    setState(() {
                      if (_noteController.text.isEmpty) {
                        _noteController.text = preset;
                      } else if (!_noteController.text.contains(preset)) {
                        _noteController.text = '${_noteController.text}, $preset';
                      }
                    });
                  },
                  label: Text(preset, style: TextStyle(fontSize: 11, color: isSelected ? Colors.white : AppColors.textPrimaryLight)),
                  backgroundColor: AppColors.surfaceLightChip,
                  selectedColor: AppColors.primary,
                  showCheckmark: false,
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
                  side: BorderSide.none,
                  padding: const EdgeInsets.symmetric(horizontal: 4, vertical: 0),
                );
              }).toList(),
            ),

            const SizedBox(height: 16),

            Row(
              mainAxisAlignment: MainAxisAlignment.end,
              children: [
                OutlinedButton(
                  onPressed: widget.onDismiss,
                  style: OutlinedButton.styleFrom(
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                  ),
                  child: const Text('Batal'),
                ),
                const SizedBox(width: 8),
                ElevatedButton(
                  onPressed: () {
                    widget.onSaveNotes(_noteController.text.trim());
                    widget.onDismiss();
                  },
                  style: ElevatedButton.styleFrom(
                    backgroundColor: AppColors.primary,
                    foregroundColor: Colors.white,
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                    elevation: 0,
                  ),
                  child: const Text('Simpan Catatan'),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
